hcatalog/core/src/main/java/org/apache/hive/hcatalog/cli/SemanticAnalysis/HCatSemanticAnalyzer.java

```diff
@@ -348,8 +348,8 @@ protected void authorizeDDLWork(HiveSemanticAnalyzerHookContext cntxt, Hive hive
 
     AlterTableDesc alterTable = work.getAlterTblDesc();
     if (alterTable != null) {
-      Table table = hive.getTable(SessionState.get().getCurrentDatabase(),
-          Utilities.getDbTableName(alterTable.getOldName())[1], false);
+      final String tableName = Utilities.getTableName(alterTable.getOldName()).getTable();
```

Comment:
I guess we can't add `of` methods to `TableName`; but I feel like putting this into "Utilities" is kinda unfortunate.
How about: `HiveTableName.of(...)`
If you need any utilities you can probably make HiveTableName extend TableName - not sure if that will be needed or not